### ISSUE TITLE
[Core][Feature request] Promote `actor_handle._actor_id` to public API

### ISSUE THREAD

**Issue by cadedaniel** [ray-project/ray#32638]
We should promote `_actor_id` to public API and also make it a usable string, e.g. `repr` does not add anything except the ID.

```
>>> a._actor_id
ActorID(811a6143d62d08e4d6e447c21c000000)
# This should change to just return '811a6143d62d08e4d6e447c21c000000'
```

Example use case is in combination with the State API:
```
import ray

ray.init()

@ray.remote
class A:
    pass

a = A.remote()

# Use here instead of this hack
obj_ref = str(a).replace('Actor(A, ', '').replace(')', '')

from ray.experimental.state.api import get_actor
print(get_actor(id=obj_ref))
```

**Comment by pcmoritz** [comment]
Yeah, exposing `actor_handle.actor_id()` seems like a no-brainer here. I think it should actually be `actor_handle.get_actor_id()` and return a string to be consistent with the APIs in https://docs.ray.io/en/master/ray-core/api/doc/ray.runtime_context.RuntimeContext.html

Any thoughts @ericl or @scv119 ?

**Comment by ericl** [comment]
Wouldn't this conflict with a method called 'get_actor_id' on the actual actor?

**Comment by pcmoritz** [comment]
Great point! So maybe be consistent with the other "hidden" methods then, call it `actor_handle.__ray_actor_id__()` then and still make it public? Or keep it like it is and make it public?

**Comment by ericl** [comment]
__ray_terminate__ currently launches a remote method, so we should probably make this syntactically different. I'd vote to make the current thing public.

**Comment by pcmoritz** [comment]
Alright, sounds good to me! This has the advantage that it doesn't need migrations. We can just say in the docs `even though this starts with an underscore to not clash with actor methods, this is a public interface` or something like that :)

**Comment by pcmoritz** [comment]
I think the main action item of this issue will be to add `ActorHandle` to https://docs.ray.io/en/latest/ray-core/package-ref.html and make sure this is documented.

**Comment by scv119** [comment]
Would an alternative is to add an helper method `ray.get_actor_id(actor_handle)`?

**Comment by krtkvrm** [comment]
can I take this up ?

**Comment by martinlhw** [comment]
Update after internal discussion-- if this were to be picked up, the preferred direction is a standalone public function, for example `ray.get_actor_id(actor_handle)`, that returns the actor ID as a hex string. There's also a related idea in #59405 to split ActorHandle into a private implementation class and a small public class that only exposes the intended attributes. `actor_handle._actor_id.hex()` works in the meantime.
